interview-questions.md

```diff
@@ -8,49 +8,53 @@ Then, research the question to expand on your answer. Even if you feel you have
 
 
 
 2. The JavaScript built in method .map() takes predefined parameters. What are they? Which are required and which are optional?
 
-  Your answer:
+  Your answer: The parameters that .map() takes is an anonymous function and values such as value, index, and array. Value is required but index and array are optional
 
-  Researched answer:
+  Researched answer: The .map bulit-in method takes of 3 arguments value, index and array. Value is required as it is the only parameter needed when simplifying the syntax. Index, and array are not required and are used more so for additional information. 
 
 
 
 3. What is the difference between map and filter?
 
-  Your answer:
+  Your answer: .map is a higher order function that returns a new array that is the same length as the original, were as .filter is a higher order function that returns an array that has filtered out information we do not want and give only the information we asked for. For example, .map would be used to create a function that was take an array of numbers and return a new array with every number multiplied by 3. We would use .filter if we needed to create a function that would take an array of numbers and return an array with only the even numbers from the original array. 
 
-  Researched answer:
+  Researched answer: .map and .filter are both methods that create new arrays, however, .map acts on every element within the array and returns an array with these elements being changed in someway. .filter creates a new array by removing unwanted elements or elements that do not belong. 
 
 
 
 4. What is the difference between a function and a method?
 
-  Your answer:
+  Your answer: A function is an action that can be called upon, given an input and return an output. Methods are actions that are called within a function and act on the arguments within the function. 
 
-  Researched answer:
+  Researched answer: A function is code that can be reused and called whenever needed. This is helpful as it does not required you to rewrite the same code over and over. Functions and Methods similar in that they perform tasks but methods differ in that they are associated with objects. 
 
 
 
 5. What is object destructuring?
 
-  Your answer:
+  Your answer: Object destructing is when you declare a variable and assign it the pathway to reach a specific key of a value within an object
 
-  Researched answer:
+  Researched answer: Object destructing is a way to assign properties/keyes of an object to a variable. 
 
 
 
 6. STRETCH: What is hoisting in JavaScript?
 
-  Your answer:
+  Your answer: hoisting in Javascript is to pull or move information in your code. 
 
-  Researched answer:
+  Researched answer: Hoisting in JavaScript is the act of moving declaration to the top of the current function. It allows for variables to be used before they are declared. 
```

Comment:
Fantastic improvement over last assessment on the interview questions!  However, you could still add examples to show a deeper understanding.  I would encourage you to stretch yourself in this area.  You can do it - you know the info and concepts. Practicing explaining it all before the actual interview will make such a huge difference.